docs/integrating-with-relay.md

```diff
@@ -0,0 +1,179 @@
+RUN pip --no-cache-dir install "https://packages.nebula.puppet.net/sdk/support/python/v1/nebula_sdk-1-py3-none-any.whl"
+```
+
+The SDK itself does not yet have friendly documentation; the source code is at [puppetlabs/nebula-sdk](https://github.com/puppetlabs/nebula-sdk/tree/master/support/python/src/nebula_sdk) for the brave of heart.
+
+### Writing entrypoint code
+
+Whichever route you've chosen, at this point you've got the setup for a Relay-friendly container. The last part is probably the most interesting, because it's where you need to write enough code to make the container do your bidding. We'll handle the examples for Steps and Triggers slightly differently, since there are different APIs you'll be working with.
+
+#### Trigger entrypoints and webhook API
+
+Under the hood, when a workflow references a webhook trigger, the Relay app will prompt the user to register the webhook on the remote service. Once it's set up, Relay associates all incoming requests to their associated trigger containers. When a request comes in, Knative launches the appropriate container and connects the incoming payload to the entrypoint, a http handler. The container's job is to:
+
+* Start a webserver on port 8080
+* Handle a POST request and decode the incoming payload
+* Decide whether to run the rest of the workflow
+* If so, map values from the request payload onto workflow parameters
+* Finally, send this mapping back into the Relay service as an event
+
+The Relay Python SDK is by far the easiest way to do this. The [Integration template repository](https://github.com/relay-integrations/template/) has a simple starting point, and there are full-featured examples for [Dockerhub push events](https://github.com/relay-integrations/relay-dockerhub/blob/master/actions/triggers/image-pushed/handler.py) and [new Pagerduty incidents](https://github.com/relay-integrations/relay-pagerduty/blob/master/actions/triggers/incident-triggered/handler.py). Check out the [Relay integrations on Github](https://github.com/relay-integrations/) for more ideas.
+
+#### Step entrypoints
+
+Steps have a relatively easy lot in life. They run, do some work to advance the state of the workflow, and exit. In many cases, a step can use existing scripts which you modify only enough to take advantage of the Relay service API. The [Getting started](./getting-started.md) shows an example workflow that uses the `input` map to access this API for retrieving and setting keys and values to pass data through the workflow. For more advanced uses, you'll probably need a script that uses the [ni utility](./cli/ni.md) or its [python SDK equivalent](https://github.com/puppetlabs/nebula-sdk/blob/master/support/python/src/nebula_sdk/interface.py) to get and set data. Formally, a step container running in Relay can expect:
+
+* It will be executed without persistent storage
+* Only keys declared in the step's `spec` map in the workflow will be available in the metadata service
```

Comment:
The metadata service is mentioned in passing several times, but maybe worth expanding on eventually?